cmd/fleet/handleCheckin.go

```diff
@@ -315,33 +317,32 @@ func parsePolicy(ctx context.Context, bulker bulk.Bulk, agentId string, action A
 		}
 	}
 
-	var ok bool
-	if ok = setMapObj(actionObj, agent.DefaultApiKey, "policy", "outputs", "default", "api_key"); !ok {
-		ok = setMapObj(actionObj, agent.DefaultApiKey, "policy", "config", "default", "api_key")
+	if ok := setMapObj(actionObj, agent.DefaultApiKey, "outputs", "default", "api_key"); !ok {
+		log.Debug().Msg("Cannot inject api_key into policy")
 	}
 
-	dataJSON := []byte(action.Data)
-	if ok {
-		// Reserialize
-		var err error
-		if dataJSON, err = json.Marshal(actionObj); err != nil {
-			return nil, err
-		}
-	} else {
-		log.Debug().Msg("Cannot inject api_key into action")
+	dataJSON, err := json.Marshal(struct {
+		Policy map[string]interface{} `json:"policy"`
+	}{actionObj})
+	if err != nil {
+		return nil, err
 	}
 
 	resp := ActionResp{
 		AgentId:   agent.Id,
-		CreatedAt: action.CreatedAt,
+		CreatedAt: policy.Timestamp,
 		Data:      dataJSON,
-		Id:        action.Id,
-		Type:      action.Type,
+		Id:        idForPolicy(policy.PolicyId, policy.RevisionIdx, policy.CoordinatorIdx),
+		Type:      TypePolicyChange,
 	}
 
 	return &resp, nil
 }
 
+func idForPolicy(policyId string, revIdx int64, coordIdx int64) string {
```

Comment:
nit:
this and the code above seem like a good candidate for the policyId wrapper that supports Parse() and String()